### PR TITLE
[Network] #1 Renaming message and messagetypeid

### PR DESCRIPTION
**Pull request by nikkolasg**
This issue fixes https://github.com/dedis/cothority/issues/766, and is part of https://github.com/dedis/cothority/issues/722.

### CODE REVIEW

File: network/dispatch.go [hunk 1]
```diff
@@ -16,31 +16,31 @@ import (
...
+	// be called for each message of type msgType. It's a shorter way of
 	// registering a Processor.
-	RegisterProcessorFunc(PacketTypeID, func(*Packet))
+	RegisterProcessorFunc(MessageTypeID, func(*Envelope))
 	// Dispatch will find the right processor to dispatch the packet to. The id
 	// is the identity of the author / sender of the packet.
 	// It can be called for example by the network layer.
 	// If no processor is found for this message type, then an error is returned
-	Dispatch(packet *Packet) error
+	Dispatch(packet *Envelope) error
```
Comment (ineiti) [inline network/dispatch.go]: should this be `envelope *Envelope`?

File: network/dispatch.go [hunk 2]
```diff
@@ -16,31 +16,31 @@ import (
...
-// packets. It is used in conjunction with Dispatcher:
+// messages. It is used in conjunction with Dispatcher:
 // A processor must register itself to a Dispatcher so the Dispatcher will
-// dispatch every messages to the Processor asked for.
+// dispatch every messages asked for to the Processor.
 type Processor interface {
-	// Process takes a received Packet.
-	Process(packet *Packet)
+	// Process takes a received Envelope.
+	Process(packet *Envelope)
```
Comment (ineiti) [inline network/dispatch.go]: `envelope *Envelope`?

File: network/encoding.go [hunk 3]
```diff
@@ -45,7 +45,7 @@ func (pId PacketTypeID) String() string {
 }
 
 // Equal returns true if pId is equal to t
-func (pId PacketTypeID) Equal(t PacketTypeID) bool {
+func (pId MessageTypeID) Equal(t MessageTypeID) bool {
```
Comment (ineiti) [inline network/encoding.go]: `mId MessageTypeID`

File: network/encoding.go [hunk 4]
```diff
@@ -128,39 +128,39 @@ func DefaultConstructors(suite abstract.Suite) protobuf.Constructors {
 
 // Error returns the error that has been encountered during the unmarshaling of
 // this message.
-func (am *Packet) Error() error {
+func (am *Envelope) Error() error {
```
Comment (ineiti) [inline network/encoding.go]: `en *Envelope`

File: network/encoding.go [hunk 5]
```diff
@@ -128,39 +128,39 @@ func DefaultConstructors(suite abstract.Suite) protobuf.Constructors {
...
-		types: make(map[PacketTypeID]reflect.Type),
+		types: make(map[MessageTypeID]reflect.Type),
 		lock:  sync.Mutex{},
 	}
 }
 
 // get returns the reflect.Type corresponding to the registered PacketTypeID
 // an a boolean indicating if the type is actually registered or not.
-func (tr *typeRegistry) get(id PacketTypeID) (reflect.Type, bool) {
+func (tr *typeRegistry) get(id MessageTypeID) (reflect.Type, bool) {
```
Comment (ineiti) [inline network/encoding.go]: `id` -> `mid`

File: network/encoding.go [hunk 6]
```diff
@@ -170,26 +170,26 @@ var registry = newTypeRegistry()
...
+var EmptyEnvelope = Envelope{MsgType: MessageTypeID(uuid.Nil)}
 
 // global mutex for MarshalRegisteredType
 var marshalLock sync.Mutex
 
 // MarshalRegisteredType will marshal a struct with its respective type into a
 // slice of bytes. That slice of bytes can be then decoded in
 // UnmarshalRegisteredType. data must be a pointer to the message.
-func MarshalRegisteredType(data Body) ([]byte, error) {
+func MarshalRegisteredType(data Message) ([]byte, error) {
```
Comment (ineiti) [inline network/encoding.go]: `msg Message`

File: network/encoding.go [hunk 7]
```diff
@@ -213,9 +213,9 @@ func MarshalRegisteredType(data Body) ([]byte, error) {
 // UnmarshalRegisteredType returns the type, the data and an error trying to
 // decode a message from a buffer.
 // The type must be registered to the network library in order to be decodable.
-func UnmarshalRegisteredType(buf []byte, constructors protobuf.Constructors) (PacketTypeID, Body, error) {
+func UnmarshalRegisteredType(buf []byte, constructors protobuf.Constructors) (MessageTypeID, Message, error) {
 	b := bytes.NewBuffer(buf)
-	var tID PacketTypeID
+	var tID MessageTypeID
```
Comment (ineiti) [inline network/encoding.go]: `mID MessageTypeID`

File: network/network.go [hunk 8]
```diff
@@ -5,11 +5,11 @@ type Conn interface {
 	// Send a message through the connection.
 	// obj should be a POINTER to the actual struct to send, or an interface.
 	// It should not be a Golang type.
-	Send(obj Body) error
+	Send(obj Message) error
```
Comment (ineiti) [inline network/network.go]: `msg Message`

File: network/router_test.go [hunk 9]
```diff
@@ -222,7 +222,7 @@ func newNSquareProc(t *testing.T, r *Router, expect int, wg *sync.WaitGroup) *nS
 	return &nSquareProc{t, r, expect, wg, make(map[Address]bool), make(map[Address]bool), sync.Mutex{}}
 }
 
-func (p *nSquareProc) Process(pack *Packet) {
+func (p *nSquareProc) Process(pack *Envelope) {
```
Comment (ineiti) [inline network/router_test.go]: `env *Envelope`

File: network/tcp.go [hunk 10]
```diff
@@ -72,27 +72,26 @@ func NewTCPConn(addr Address) (conn *TCPConn, err error) {
 }
 
 // Receive get the bytes from the connection then decodes the buffer.
-// It returns the Packet with the Msg field decoded
-// or EmptyApplicationPacket and an error if something wrong happened.
-func (c *TCPConn) Receive() (nm Packet, e error) {
+// It returns the Envelope containing the message,
+// or EmptyEnvelope and an error if something wrong happened.
+func (c *TCPConn) Receive() (nm Envelope, e error) {
```
Comment (ineiti) [inline network/tcp.go]: `env *Envelope` - why is this a value and not a pointer?

File: network/tcp.go [hunk 11]
```diff
@@ -138,10 +137,10 @@ func (c *TCPConn) receiveRaw() ([]byte, error) {
 // Send converts the NetworkMessage into an ApplicationMessage
 // and sends it using send().
 // It returns an error if anything was wrong.
-func (c *TCPConn) Send(obj Body) error {
+func (c *TCPConn) Send(obj Message) error {
```
Comment (ineiti) [inline network/tcp.go]: `msg Message`

File: network/tcp_test.go [hunk 12]
```diff
@@ -501,7 +501,7 @@ func newSimpleMessageProc(t *testing.T) *simpleMessageProc {
 	}
 }
 
-func (smp *simpleMessageProc) Process(p *Packet) {
+func (smp *simpleMessageProc) Process(p *Envelope) {
```
Comment (ineiti) [inline network/tcp_test.go]: `e *Envelope`

File: node_test.go [hunk 13]
```diff
@@ -306,7 +306,7 @@ func (c *ServiceChannels) NewProtocol(tn *TreeNodeInstance, conf *GenericConfig)
 	return NewProtocolChannels(tn)
 }
 
-func (c *ServiceChannels) Process(packet *network.Packet) {
+func (c *ServiceChannels) Process(packet *network.Envelope) {
```
Comment (ineiti) [inline node_test.go]: `env *network.Envelope`

File: network/tcp.go (hunk 10 above)
Comment (nikkolasg) [inline network/tcp.go]: Well there is no "why", we just did not do it like that ;)
Feel free to issue that or add to existing network issue.

File: network/dispatch_test.go [hunk 14]
```diff
@@ -49,10 +49,10 @@ func TestBlockingDispatcher(t *testing.T) {
 	}
 
 	var found bool
-	dispatcher.RegisterProcessorFunc(basicMessageType, func(p *Packet) {
+	dispatcher.RegisterProcessorFunc(basicMessageType, func(p *Envelope) {
```
Comment (ineiti) [inline network/dispatch_test.go]: `e *Envelope`

File: network/dispatch_test.go [hunk 15]
```diff
@@ -95,11 +95,11 @@ func TestRoutineDispatcher(t *testing.T) {
 
 func TestDefaultProcessor(t *testing.T) {
 	var okCh = make(chan bool, 1)
-	pr := defaultProcessor{func(p *Packet) {
+	pr := defaultProcessor{func(p *Envelope) {
```
Comment (ineiti) [inline network/dispatch_test.go]: `e *Envelope`

File: network/encoding.go [hunk 16]
```diff
@@ -254,32 +254,32 @@ func UnmarshalRegistered(buf []byte) (PacketTypeID, Body, error) {
...
+	env.MsgType = t
+	env.Msg = msg
 	return err
 }
 
-// NewNetworkPacket takes a Body and then constructs a
+// NewEnvelope takes a Body and then constructs a
 // Message from it. Error if the type is unknown
-func NewNetworkPacket(obj Body) (*Packet, error) {
+func NewEnvelope(obj Message) (*Envelope, error) {
```
Comment (ineiti) [inline network/encoding.go]: `msg Message`

File: network/tcp.go (hunk 10 above)
Comment (ineiti) [inline network/tcp.go]: https://github.com/dedis/onet/issues/57

File: network/tcp.go [hunk 17]
```diff
@@ -72,27 +72,26 @@ func NewTCPConn(addr Address) (conn *TCPConn, err error) {
...
 	defer func() {
 		if err := recover(); err != nil {
 			e = fmt.Errorf("Error Received message: %v\n%s", err, log.Stack())
-			nm = EmptyApplicationPacket
+			nm = EmptyEnvelope
 		}
 	}()
 
-	var am Packet
+	var am Envelope
```
Comment (ineiti) [inline network/tcp.go]: `em Envelope`

File: protocol.go [hunk 18]
```diff
@@ -169,7 +169,7 @@ func (p *messageProxyStore) getByName(name string) MessageProxy {
 	return p.defaultIO
 }
 
-func (p *messageProxyStore) getByPacketType(t network.PacketTypeID) MessageProxy {
+func (p *messageProxyStore) getByPacketType(t network.MessageTypeID) MessageProxy {
```
Comment (ineiti) [inline protocol.go]: `m network.MessageTypeID`

**Pull request merged**